api/uniter/uniter.go

```diff
@@ -433,6 +433,30 @@ func (st *State) SLALevel() (string, error) {
 	return result.Result, nil
 }
 
+// GoalState returns a GoalStateResult struct with the charm's
+// peers and related units information.
+func (c *State) GoalState() (params.StringResult, error) {
```

Comment:
We don't return a StringResult here. In general, we try (but don't always succeed) in not leaking params structs outside the api layer.
Here, it's just a simple type so easy to do. You just want to return a string, ie 

func (c *State) GoalState() (string, error) {
...
result.Results[0].Result, nil
}